File: samples/client/3_1_0_unit_test/java/src/test/java/org/openapijsonschematools/client/components/schemas/UnevaluatedpropertiesSchemaTest.java
package org.openapijsonschematools.client.components.schemas;

import org.junit.Assert;
import org.junit.Test;
import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;
import org.openapijsonschematools.client.schemas.validation.MapUtils;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.AbstractMap;

public class UnevaluatedpropertiesSchemaTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(new JsonSchemaKeywordFlags.Builder().format().build());

    @Test
    public void testWithInvalidUnevaluatedPropertiesFails() {
        // with invalid unevaluated properties
        final var schema = UnevaluatedpropertiesSchema.UnevaluatedpropertiesSchema1.getInstance();
        try {
            schema.validate(
                MapUtils.makeMap(
                    new AbstractMap.SimpleEntry<String, String>(
                        "foo",
                        "fo"
                    )
                ),
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }

    @Test
    public void testWithNoUnevaluatedPropertiesPasses() throws ValidationException {
        // with no unevaluated properties
        final var schema = UnevaluatedpropertiesSchema.UnevaluatedpropertiesSchema1.getInstance();
        schema.validate(
            MapUtils.makeMap(
            ),
            configuration
        );
    }

    @Test
    public void testWithValidUnevaluatedPropertiesPasses() throws ValidationException {
        // with valid unevaluated properties
        final var schema = UnevaluatedpropertiesSchema.UnevaluatedpropertiesSchema1.getInstance();
        schema.validate(
            MapUtils.makeMap(
                new AbstractMap.SimpleEntry<String, String>(
                    "foo",
                    "foo"
                )
            ),
            configuration
        );
    }
}

File: samples/client/3_1_0_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/UnevaluatedpropertiesSchemaTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class UnevaluatedpropertiesSchemaTest {
    companion object {
        val configuration = SchemaConfiguration(JsonSchemaKeywordFlags.Builder().format().build())
    }

    @Test
    fun testWithInvalidUnevaluatedPropertiesFails() {
        // with invalid unevaluated properties
        val schema = UnevaluatedpropertiesSchema.UnevaluatedpropertiesSchema1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    mapOf(
                        Pair(
                            "foo",
                            "fo"
                        )
                    ),
                    configuration
                )
            }
        )
    }

    @Test
    fun testWithNoUnevaluatedPropertiesPasses() {
        // with no unevaluated properties
        val schema = UnevaluatedpropertiesSchema.UnevaluatedpropertiesSchema1.getInstance()
        schema.validate(
            mapOf<Any?, Any?>(
            ),
            configuration
        )
    }

    @Test
    fun testWithValidUnevaluatedPropertiesPasses() {
        // with valid unevaluated properties
        val schema = UnevaluatedpropertiesSchema.UnevaluatedpropertiesSchema1.getInstance()
        schema.validate(
            mapOf(
                Pair(
                    "foo",
                    "foo"
                )
            ),
            configuration
        )
    }
}
